File: 0_liste&dict/C++/lc3396.cpp
class Solution {
    public:
        int minimumOperations(std::vector<int>& nums) {
            int nb_op = 0;
             
            while (nums.size() > std::unordered_set<int>(nums.begin(), nums.end()).size()) {
    
                if (nums.size() >= 3) {
                    nums = std::vector<int>(nums.begin() + 3, nums.end());
                } else {
                    nums.clear();
                }
                nb_op++;
            }
            
            return nb_op;
        }
    };

File: 0_liste&dict/Py/lc3396.py
from typing import List


class Solution:
    def minimumOperations(self, nums: List[int]) -> int:
        nb_op = 0
        while len(nums) > len(set(nums)):
            nums = nums[3:]
            nb_op += 1
        return nb_op
    

# The loop runs O(n) times in the worst case.
# Each iteration create a new set (O(n)) and slice the list (O(n)).
# The total time complexity is O(n^2).
# The space complexity is O(n) for the set.